hydra-node/test/Hydra/OptionsSpec.hs

```diff
@@ -324,6 +325,23 @@ spec = parallel $
             { chainConfig = Cardano (defaultCardanoChainConfig & #chainBackendOptions .~ Blockfrost (BlockfrostOptions "blockfrost-project.txt"))
             }
 
+    it "parses --network into related tx ids" $ do
+      -- NOTE: we should be able to parse both upper and lower case network names
+      let networks = ["Mainnet", "preview", "Preprod"]
+      forM_ networks $ \network -> do
+        case parseNetworkTxIds network of
+          Left err ->
+            err `shouldSatisfy` \case
+              "Failed to find released hydra-node version in networks.json." -> True
+              "Missing hydra-node revision." -> True
+              _ -> False
```

Comment:
Is this testing the implementation of the command line parser with the main function that implements the parser? :)

Also, matching strings is a bit smelly here.

Also, this test actually tests the parsed transaction ids only when we are running it on a release (not in development).

Can we please change this to:
- Lower requirement to the parser here, i.e. that it just returns _some_ tx ids
- Unit test the `parseNetworkTxIds`
  - given good/bad network names
  - given good/bad versions (`git describe` outputs we expect to be usable or not)

Reply:
I've removed parts that assert on exact string when failing to parse and added a property test to make sure `parseNetworkTxIds` works with known networks.